### DISCUSSION TITLE
Hash for vcpkg binary caching

### DISCUSSION THREAD

**Discussion by johnwason** [microsoft/vcpkg#20759] [Q&A] [answered]
Github Actions provides a cache action (actions/cache@v2) that works pretty well to store the binary build artifacts output by vcpkg. This has greatly sped up repeat builds. However, the github actions cache requires a "restore key" that will change when the source files are updated. Right now, it isn't clear how to generate this restore key. vcpkg uses a complex set up hashes to determine when to use a cached artifact, or when to rebuild. Is there a way to access (or easily generate) a hash from the current vcpkg directory, so that it can be used as the "restore key" for the github actions cache?

**Comment by johnwason** [comment, marked as answer]
I discovered the best way is to use a dry run, and save the output to file. This can then be cached. See https://github.com/johnwason/vcpkg-action for an action that uses this method.

**Comment by mathisloge** [comment]
Note that the actions/cache is limited to IIRC 5GB. So it will run pretty fast out of space. I recommend going the way to use the real binary caching provided by vcpkg and described in https://github.com/microsoft/vcpkg/blob/master/docs/users/binarycaching.md#github-packages

**Reply by johnwason** [reply]
Thanks for the tip @mathisloge . I haven't run into a caching size limit yet. My projects are  not using all that many vcpkg libraries, and I have been using the -release triplets. Most of the installed size of these libraries on disk are *.pdb files, so it helps a lot to not build them if they aren't being used.
